pep-0698.rst

```diff
@@ -0,0 +1,439 @@
+
+    def parent_callsite(parent: Parent) -> None:
+        # If we pass a Child instance we’ll now run Parent.new_foo - likely a bug
+        parent.new_foo(1)
+
+    def child_callsite(child: Child) -> None:
+        # We probably wanted to invoke new_foo here. Instead, we forked the method
+        child.foo(1)
+
+
+This code will type check, but there are two potential sources of bugs:
+
+- If we pass a ``Child`` instance to the parent_callsite function, it will
+  invoke the implementation in ``Parent.new_foo``. rather than ``Child.foo``.
+  This is probably a bug - we presumably would not have written ``Child.foo`` in
+  the first place if we didn’t need custom behavior.
+- Our system was likely relying on ``Child.foo``  behaving in a similar way to
+  ``Parent.foo``. But unless we catch this early, we have now forked the
+  methods, and future refactors it is likely no one will realize that major
+  changes to the behavior of new_foo likely require updating ``Child.foo`` as
+  well, which could lead to major bugs later.
+
+The incorrectly-refactored code is type-safe, but is probably not what we
+intended and could cause our system to behave incorrectly. The bug can be
+difficult to track down because our new code likely does execute without
+throwing exceptions. Tests are less likely to catch the problem, and silent
+errors can take longer to track down in production.
+
+We are aware of several production outages in multiple typed codebases caused by
+such incorrect refactors. This is our primary motivation for adding an ``@override``
+decorator to the type system, which lets developers express the relationship
+between ``Parent.foo`` and ``Child.foo`` so that type checkers can detect the problem.
+
+
+Subclass Implementations Become More Explicit
+---------------------------------------------
+
+We believe that explicit overrides will make unfamiliar code easier to read than
+implicit overrides. A developer reading the implementation of a subclass that
+uses ``@override``  can immediately see which methods are overriding
+functionality in some base class; without this decorator, the only way to
+quickly find out is using a static analysis tool.
+
+
+Why Not Rely on Integrated Development Environments?
+----------------------------------------------------
+
+Modern Integrated Development Environments (IDEs) often provide the ability to
+automatically update subclasses when renaming a method. But we view this as
+insufficient for several reasons:
+
+- If a codebase is split into multiple projects, an IDE will not help and the
+  bug appears when upgrading dependencies. Type checkers are a fast way to catch
+  breaking changes in dependencies.
+- Not all developers use such IDEs. And library maintainers, even if they do use
+  an IDE, should not need to assume pull request authors use the same IDE. We
+  prefer being able to detect problems in continuous integration without
+  assuming anything about developers’ choice of editor.
+
```

Comment:
You could also list this under a `Rejected Ideas` section near the end, following the traditional PEP template